cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedCall.qll

```diff
@@ -368,26 +377,48 @@ class TranslatedSideEffects extends TranslatedElement, TTranslatedSideEffects {
     operandTag = addressOperand() and
     result = getPrimaryInstructionForSideEffect(OnlyInstructionTag())
   }
+}
+
+class TranslatedCallAllocationSideEffects extends TranslatedAllocationSideEffects {
+  override CallAllocationExpr expr;
 
   override Instruction getPrimaryInstructionForSideEffect(InstructionTag tag) {
     tag = OnlyInstructionTag() and
     result = getTranslatedExpr(expr).getInstruction(CallTag())
   }
+}
 
-  /**
-   * Gets the `TranslatedFunction` containing this expression.
-   */
-  final TranslatedFunction getEnclosingFunction() {
-    result = getTranslatedFunction(expr.getEnclosingFunction())
+class TranslatedNewAllocationSideEffects extends TranslatedAllocationSideEffects {
+  override NewAllocationExpr expr;
```

Comment:
Don't we need to support `NewArrayAllocationExpr` too?

Reply:
You're correct. I forgot about it the first time around, but 291df97cd96ae264b7f5a402636baf59ddf7f965 adds a `TranslatedNewArrayAllocationSideEffects` class that does just that.